openeo/rest/userfile.py

```diff
@@ -0,0 +1,63 @@
+import typing
+from typing import Any, Dict, Union
+import os
+from pathlib import Path
+from openeo.util import ensure_dir
+
+if typing.TYPE_CHECKING:
+    # Imports for type checking only (circular import issue at runtime).
+    from openeo.rest.connection import Connection
+
+
+class UserFile:
+    """Represents a file in the user-workspace of openeo."""
+
+    def __init__(self, path: str, connection: 'Connection', metadata: Dict[str, Any] = None):
+        self.path = path
+        self.metadata = metadata or {"path": path}
+        self.connection = connection
+
+    def __repr__(self):
+        return '<{c} file={i!r}>'.format(c=self.__class__.__name__, i=self.path)
+
+    def _get_endpoint(self) -> str:
+        return "/files/{}".format(self.path)
+
+    def download(self, target: Union[Path, str] = None) -> Path:
+        """
+        Downloads a user-uploaded file to the given location.
+
+         :param target: download target path. Can be an existing folder 
+             (in which case the file name advertised by backend will be used) 
+             or full file name. By default, the working directory will be used.
+        """
+        # GET /files/{path}
+        response = self.connection.get(self._get_endpoint(), expected_status=200, stream=True)
+
+        target = Path(target or Path.cwd()) 
+        if target.is_dir():
+            target = target / os.path.basename(self.path)
```

Comment:
```suggestion
            target = target / self.path.name
```

And related to note higher: user might want to keep original folder structure, in which case it should be:
```suggestion
            target = target / self.path
```

Reply:
Does this create missing folders automatically?